providers/cluster.rb

```diff
@@ -0,0 +1,259 @@
+#
+# Cookbook Name:: rabbitmq
+# Provider:: cluster
+#
+# Author: Sunggun Yu <sunggun.dev@gmail.com>
+# Copyright (C) 2015 Sunggun Yu
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+include Chef::Mixin::ShellOut
+
+use_inline_resources
+
+# Get ShellOut
+def get_shellout(cmd)
+  sh_cmd = Mixlib::ShellOut.new(cmd)
```

Comment:
There is always a concern for timeouts on shellout, especially for cluster operations. We should probably note that or make it configurable.